src/AzureAppConfigurationImpl.ts

```diff
@@ -291,17 +417,73 @@ export class AzureAppConfigurationImpl implements AzureAppConfiguration {
                 this.#refreshTimer.backoff();
                 throw error;
             }
+            return Promise.resolve(true);
+        }
 
-            // successfully refreshed, run callbacks in async
-            for (const listener of this.#onRefreshListeners) {
-                listener();
+        return Promise.resolve(false);
+    }
+
+    /**
+     * Refresh feature flags.
+     * @returns true if feature flags are refreshed, false otherwise.
+     */
+    async #refreshFeatureFlags(): Promise<boolean> {
+        // if still within refresh interval/backoff, return
+        if (!this.#featureFlagRefreshTimer.canRefresh()) {
+            return Promise.resolve(false);
+        }
+
+        // check if any feature flag is changed
+        let needRefresh = false;
+        for (const selector of this.#featureFlagSelectors) {
+            const listOptions: ListConfigurationSettingsOptions = {
+                keyFilter: `${featureFlagPrefix}${selector.keyFilter}`,
+                labelFilter: selector.labelFilter,
+                pageEtags: selector.pageEtags

```

Comment:
How does the SDK behave when we pass empty (list of page etags (or null)? I'm imagining a scenario where the first request to load feature flags returned 0 results.

Reply:
> How does the SDK behave when we pass empty (list of page etags (or null)? 

SDK client will send requests without adding etag related headers. See:
https://github.com/Azure/azure-sdk-for-js/blob/6c4117cf31e82aedcf2ecffe079f1899abfbe5dd/sdk/appconfiguration/app-configuration/src/appConfigurationClient.ts#L341

> a scenario where the first request to load feature flags returned 0 results.

When it returns 0 results, the page etag is not empty. What would be the concern?

Reply:
Ah I forgot that an empty page with 0 kv still has a valid etag.